src/sidebar/services/test/annotations-exporter-test.js

```diff
@@ -256,4 +256,88 @@ ${isoDate},http://example.com,My group,Annotation,,John Doe,Annotation text,"tag
       );
     });
   });
+
+  describe('buildHTMLExportContent', () => {
+    it('throws error when empty list of annotations is provided', () => {
+      assert.throws(
+        () => exporter.buildHTMLExportContent([]),
+        'No annotations to export',
+      );
+    });
+
+    it('generates HTML content with expected annotations', () => {
+      const isoDate = baseAnnotation.created;
+      const annotations = [
+        {
+          ...baseAnnotation,
+          user: 'acct:jane@localhost',
+          tags: ['foo', 'bar'],
+        },
+        {
+          ...baseAnnotation,
+          ...newReply(),
+          target: targetWithSelectors(
+            quoteSelector('includes <p>HTML</p> tags'),
+            pageSelector(23),
+          ),
+        },
+        {
+          ...baseAnnotation,
+          tags: [],
+          target: targetWithSelectors(pageSelector('iii')),
+        },
+      ];
+
+      const result = exporter.buildHTMLExportContent(annotations, {
+        groupName,
+      });
+
+      assert.equal(
+        result,
+        `<header>
+  <p><time datetime="${isoDate}">${isoDate}</time></p>
+  <p><strong>A special document</strong></p>
+  <p><a href="http://example.com" target="_blank" rel="noopener noreferrer">http://example.com</a></p>
+
+  <dl>
+    <dt>Group</dt> <dd>My group</dd>
+    <dt>Total users</dt> <dd>2</dd>
+    <dt>Users</dt> <dd>jane, bill</dd>
+    <dt>Total annotations</dt> <dd>3</dd>
+    <dt>Total replies</dt> <dd>1</dd>
+  </dl>
+</header>
+<hr />
+<section>
+  <article>
+  <strong>Annotation 1:</strong>
+  <p>${isoDate}</p>
+<p>Comment: Annotation text</p>
+<p>jane</p>
+<p>Quote: <blockquote>null</blockquote></p>
+<p>Tags: foo, bar</p>
+</article>
+<br />
+<article>
+  <strong>Annotation 2:</strong>
+  <p>${isoDate}</p>
+<p>Comment: Annotation text</p>
+<p>bill</p>
+<p>Quote: <blockquote>includes &lt;p&gt;HTML&lt;/p&gt; tags</blockquote></p>
+<p>Tags: tag_1, tag_2</p>
+<p>Page: 23</p>
+</article>
+<br />
+<article>
+  <strong>Annotation 3:</strong>
+  <p>${isoDate}</p>
+<p>Comment: Annotation text</p>
+<p>bill</p>
+<p>Quote: <blockquote>null</blockquote></p>
+<p>Page: iii</p>
+</article>
+</section>`,
```

Comment:
Produced indentation here is not great. However, it's not easy to have proper indentation of all nested elements without using a template engine.

I'm thinking on leaving it like this for now, as indentation does not affect how it renders in the browser, and it's not completely broken when reading it as plain text either.

I'll create a ticket to investigate on this later.